multimodal-search-tirg/app.py

```diff
@@ -94,11 +95,13 @@ def main(task, data_path, num_docs, batch_size, image_path, text_query, overwrit
             clean_workdir()
         f = Flow.load_config('flow-index.yml')
         with f:
-            f.index(index_generator(data_path, num_docs), batch_size=batch_size)
+            with TimeContext(f'QPS: indexing {num_docs}', logger=f.logger):
+                f.index(index_generator(data_path, num_docs), batch_size=batch_size)
     elif task == 'query':
         f = Flow.load_config('flow-query.yml')
         with f:
-            f.search(input_fn=query_generator(image_paths, text_queries), on_done=print_result)
+            with TimeContext(f'QPS: query', logger=f.logger):
```

Comment:
If we keep the QPS tracking on the blocking request we need to specify that it's for only 1 document, while here it's for `len(image_paths)`

Reply:
Sure will do here.

For the blocking case with the single document input from stdin (chinese text search), I cant really know though that it is a single document? In our context of course we know, but in code the time tracking is started outside reading from stdin so a user can feed it in multiple documents?

Reply:
I think you can assume it's only one, right? Or at least in our case (in the `.sh` script) it's only 1 doc, since it's done with this call

```
                f.search_lines(lines=[text, ], line_format='text', on_done=ppr, top_k=top_k)
```